public-to-private/run-workflow/main.go

```diff
@@ -0,0 +1,229 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-github-actions/public-to-private/run-workflow/action"
+
+	"github.com/google/go-github/v47/github"
+	"github.com/sethvargo/go-githubactions"
+	"golang.org/x/oauth2"
+)
+
+const COMPLETED_STATUS = "completed"
+
+// ConnectToGithub Sets up the github client and connects
+func ConnectToGithub(githubToken string) *github.Client {
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: githubToken},
+	)
+	tc := oauth2.NewClient(context.Background(), ts)
+	return github.NewClient(tc)
+}
+
+func StartWorkflow(client *github.Client, inputs *action.Inputs) error {
+	fmt.Printf("Starting workflow %s\n", inputs.WorkflowFile)
+	inputsMap, err := inputs.InputsToMap()
+	if err != nil {
+		return err
+	}
+	event := &github.CreateWorkflowDispatchEventRequest{
+		Ref:    inputs.Branch,
+		Inputs: inputsMap,
+	}
+	resp, err := client.Actions.CreateWorkflowDispatchEventByFileName(context.Background(), inputs.Owner, inputs.Repository, inputs.WorkflowFile, *event)
+
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode != 204 {
+		return fmt.Errorf("Failed to start the workflow, status code: %d", resp.StatusCode)
+	}
+
+	return nil
+}
+
+func GetListOfWorkflowRuns(client *github.Client, inputs *action.Inputs, workflowStartedAt time.Time) (*github.WorkflowRuns, error) {
+	opts := &github.ListWorkflowRunsOptions{
+		Branch:  inputs.Branch,
```

Comment:
Should we also filter by actor here? If the workflow dispatches are always going to be done by the same actor, it'd narrow your search result.

Reply:
Added! it is always the bot that starts it so we now always filter by the bot.